importers/algod/algod_importer.go

```diff
@@ -0,0 +1,97 @@
+package algodimporter
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+
+	"github.com/algorand/go-algorand-sdk/client/v2/algod"
+	"github.com/algorand/go-algorand/protocol"
+	"github.com/algorand/go-algorand/rpcs"
+	"github.com/algorand/indexer/data"
+	"github.com/algorand/indexer/importers"
+	"github.com/algorand/indexer/plugins"
+	"github.com/sirupsen/logrus"
+	"gopkg.in/yaml.v3"
+)
+
+const importerName = "algod"
+
+type algodImporter struct {
+	aclient *algod.Client
+	logger  *logrus.Logger
+	cfg     ImporterConfig
+	ctx     context.Context
+	cancel  context.CancelFunc
+}
+
+var algodImporterMetadata = importers.ImporterMetadata{
+	ImpName:        importerName,
+	ImpDescription: "Importer for fetching block from algod rest endpoint.",
+	ImpDeprecated:  false,
+}
+
+// New initializes an algod importer
+func New() importers.Importer {
+	return &algodImporter{}
+}
+
+func (algodImp *algodImporter) Metadata() importers.ImporterMetadata {
+	return algodImporterMetadata
+}
+
+func (algodImp *algodImporter) Init(ctx context.Context, cfg plugins.PluginConfig, logger *logrus.Logger) error {
+	algodImp.ctx, algodImp.cancel = context.WithCancel(ctx)
+	algodImp.logger = logger
+	if err := algodImp.unmarhshalConfig(string(cfg)); err != nil {
+		return fmt.Errorf("connect failure in unmarshalConfig: %v", err)
+	}
+	var client *algod.Client
+	u, err := url.Parse(algodImp.cfg.NetAddr)
+	if err != nil {
+		return err
+	}
+	if u.Scheme != "http" && u.Scheme != "https" {
+		algodImp.cfg.NetAddr = "http://" + algodImp.cfg.NetAddr
+		algodImp.logger.Warnf("Plugin config has missing or invalid URL scheme")
```

Comment:
I think it's ok if people use like `127.0.0.1:8080` as their NetAddr, so I would go with an info log instead of warning. I guess the main usefulness of the log is to say that we've defaulted you to http instead of https.

```suggestion
		algodImp.logger.Infof("Algod Importer added http prefix to NetAddr: %s", algodImp.cfg.NetAddr)
```